tests/test_operations.py

```diff
@@ -43,3 +43,213 @@ def test_failing_concat():
+        y_y=np.array([1, 3, 7, 3, 7]),
+        Ms=me.Ms([2, 3, 3, 3.5, 3.5]),
+        A=me.A([22, 33, 33.55, 33, 33.55]),
+    )
+    assert np.all(ec_merged_1.x == ec_check_1.x)
+    assert np.all(ec_merged_1.y_x == ec_check_1.y_x)
+    assert np.all(ec_merged_1.y_y == ec_check_1.y_y)
+    assert ec_merged_1.Ms == ec_check_1.Ms
+    assert ec_merged_1.A == ec_check_1.A
+
+    ec_merged_2 = me.merge(ec_1, ec_2, on=["x", "y"], how="inner")
+    ec_check_2 = me.io.EntityCollection(
+        x=np.array([2, 3, 3]),
+        y=np.array([1, 3, 3]),
+        Ms=me.Ms([2, 3, 3.5]),
+        A=me.A([22, 33, 33]),
+    )
+    assert np.all(ec_merged_2.x == ec_check_2.x)
+    assert ec_merged_2.Ms == ec_check_2.Ms
+    assert ec_merged_2.A == ec_check_2.A
+
+
+def test_merge_inner_overlap():
+    """Test inner merge function with overlapping columns."""
+    ec_1 = me.io.EntityCollection(x=[1, 2, 3], Ms=me.Ms([1, 2, 3]))
+    ec_2 = me.io.EntityCollection(
+        x=[2, 3, 4],
+        Ms=me.Ms([22, 33, 44]),
+    )
+    ec_merged_1 = me.merge(ec_1, ec_2, on="x")
+    ec_check_1 = me.io.EntityCollection(
+        x=np.array([2, 3]),
+        Ms_x=me.Ms([2, 3]),
+        Ms_y=me.Ms([22, 33]),
+    )
+    assert np.all(ec_merged_1.x == ec_check_1.x)
+    assert ec_merged_1.Ms_x == ec_check_1.Ms_x
+    assert ec_merged_1.Ms_y == ec_check_1.Ms_y
+
+    ec_merged_2 = me.merge(ec_1, ec_2, on="x", suffixes=("_1", "_2"))
+    ec_check_2 = me.io.EntityCollection(
+        x=np.array([2, 3]),
+        Ms_1=me.Ms([2, 3]),
+        Ms_2=me.Ms([22, 33]),
+    )
+    assert np.all(ec_merged_2.x == ec_check_2.x)
+    assert ec_merged_2.Ms_x == ec_check_2.Ms_x
+    assert ec_merged_2.Ms_y == ec_check_2.Ms_y
+
+    with pytest.raises(ValueError):
+        me.merge(ec_1, ec_2, on="x", suffixes=(None, None))
+
+
+def test_merge_inner_overlap_empty():
+    """Test inner merge function with overlapping columns.
+
+    If merged without specifying the attribute `on`, the merge is calculated on
+    all the common columns. We generated a entity collection corresponding to an empty
+    DataFrame and we check that the attributes have disappeared.
+    """
+    ec_1 = me.io.EntityCollection(x=[1, 2, 3], Ms=me.Ms([1, 2, 3]))
+    ec_2 = me.io.EntityCollection(
+        x=[2, 3, 4],
+        Ms=me.Ms([22, 33, 44]),
+    )
+
+    ec_merged_empty = me.merge(ec_1, ec_2)
+    assert not hasattr(ec_merged_empty, "x")
+    assert not hasattr(ec_merged_empty, "Ms")
+
+
+def test_merge_inner_different_units():
+    """Test inner merge function with different units."""
+    ec_1 = me.io.EntityCollection(
+        x=[1, 2, 3] * u.m,
+        Ms=me.Ms([1, 2, 3]),
+    )
+    ec_2 = me.io.EntityCollection(
+        x=[2000, 3000, 4000] * u.mm,
+        Ms=me.Ms([22, 33, 44]),
+    )
+    ec_merged = me.merge(ec_1, ec_2, on="x")
+    ec_check = me.io.EntityCollection(
+        x=[2, 3] * u.m,
+        Ms_x=me.Ms([2, 3]),
+        Ms_y=me.Ms([22, 33]),
+    )
```

Comment:
If, in addition, we turn this around, i.e. `me.merge(ec_2, ec_1, on="x")`, what units do we expect for x?